.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,40 @@
+# template from https://github.com/JamesIves/github-pages-deploy-action/tree/dev
+
+name: CI
+
+run-name: Deployment Triggered - ${{ github.event.workflow_run.conclusion == 'success'}}
+
+on:
+  workflow_run:
+    workflows: ["Run tests"]
+    types:
+      - completed
+
+jobs:
+  build-and-deploy:
+    #concurrency: ci-${{ github.ref }} # Recommended if you intend to make multiple deployments in quick succession.
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Install and Build 🔧
+        run: |
+          npm install
+          npm run-script build
+
+      - name: Deploy 🚀
+        uses: JamesIves/github-pages-deploy-action@v4
+        with:
+          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+          folder: dist/word-weaver # The folder the action should deploy.
+
+  on-failure:
+    name: Validation Failed
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'failure' }}
+    steps:
+      - run: echo "Tests conclusion was failure. Deployment not ran."
```

Comment:
Nice to have this on-failure step, I like having a clear and explicit log like this.
Typo: "Deployment not ran" is not grammatical, maybe "Deployment cancelled" or "skipped" or something?